fuse/fuse.cpp

```diff
@@ -37,25 +38,78 @@
 #include "namespace.h"
 #endif
 
-int run_in_fuse(
-	const std::string& daemon_path,
-	const std::string& working_dir,
-	const std::string& json,
-	bool use_stdin_file,
-	std::string& result_json)
-{
-	if (0 != chdir(working_dir.c_str())) {
-		std::cerr << "chdir " << working_dir << ": " << strerror(errno) << std::endl;
-		return 1;
-	}
-
+bool json_as_struct(std::string json, json_args& result) {
 	JAST jast;
 	if (!JAST::parse(json, std::cerr, jast))
+		return false;
+
+	for (auto &x : jast.get("command").children)
+		result.command.push_back(x.second.value);
+
+	for (auto &x : jast.get("environment").children)
+		result.environment.push_back(x.second.value);
+
+	for (auto &x : jast.get("visible").children)
+		result.visible.push_back(x.second.value);
+
+	result.directory = jast.get("directory").value;
+	result.stdin_file = jast.get("stdin").value;
+
+#ifdef __linux__
+	result.isolate_network = jast.get("isolate-network").kind == JSON_TRUE;
+
+	std::string hostname = jast.get("hostname").value;
+	result.hostname = !hostname.empty() ? hostname : "build";
+
+	std::string domainname = jast.get("domainname").value;
+	result.domainname = !domainname.empty() ? domainname : "local";
+
+	std::string userid = jast.get("user-id").value;
+	result.userid = !userid.empty() ? std::stoi(userid) : geteuid();
+
+	std::string groupid = jast.get("group-id").value;
+	result.groupid = !groupid.empty() ? std::stoi(groupid) : getegid();
+
+	for (auto &x : jast.get("mount-ops").children) {
+		result.mount_ops.push_back({
```

Comment:
What happens if you use `emplace_back` instead? Does it work? If so, use that.

Reply:
```
++ -std=c++11 -Wall -O2 -DVERSION=v0.21.0-alpha-52-g4ea3d9a-dirty -Iutf8proc -Igopt -Icommon fuse/client.cpp fuse/fuse.cpp fuse/namespace.cpp common/jlexer.o common/execpath.o common/jparser.o common/json5.o common/lexint.o common/location.o common/shell.o common/unlink.o common/utf8.o common/whereami.o -o lib/wake/fuse-wake 
fuse/fuse.cpp: In function ‘bool json_as_struct(std::string, json_args&)’:
fuse/fuse.cpp:76:4: error: no matching function for call to ‘std::vector<mount_op>::emplace_back(<brace-enclosed initializer list>)’
   76 |   });
      |    ^
In file included from /usr/include/c++/10/vector:72,
                 from fuse/fuse.h:23,
                 from fuse/fuse.cpp:32:
/usr/include/c++/10/bits/vector.tcc:109:7: note: candidate: ‘void std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {}; _Tp = mount_op; _Alloc = std::allocator<mount_op>]’
  109 |       vector<_Tp, _Alloc>::
      |       ^~~~~~~~~~~~~~~~~~~
/usr/include/c++/10/bits/vector.tcc:109:7: note:   candidate expects 0 arguments, 1 provided
make: *** [Makefile:56: lib/wake/fuse-wake] Error 1
```